lib/puppet/face/config.rb

```diff
@@ -129,4 +129,44 @@
       nil
     end
   end
+
+  action(:delete) do
+    summary _("Delete a Puppet setting.")
+    arguments _("(<setting>")
+    #TRANSLATORS 'main' is a specific section name and should not be translated
+    description "Deletes a setting from the specified section. (The default is the section 'main')."
+    notes <<-'EOT'
+      By default, this action deletes the configuration setting from the 'main'
+      configuration domain. Use the '--section' flags to delete settings from other
+      configuration domains.
+    EOT
+    examples <<-'EOT'
+      Delete the setting 'setting_name' from the 'main' configuration domain:
+
+      $ puppet config delete setting_name
+
+      Delete the setting 'setting_name' from the 'master' configuration domain:
+
+      $ puppet config delete setting_name --section master
+    EOT
+
+    when_invoked do |name, options|
+
+      path = Puppet::FileSystem.pathname(Puppet.settings.which_configuration_file)
+      Puppet::FileSystem.touch(path)
```

Comment:
Is this touch to create the file if it doesn't exist?  Should we be doing that?

Reply:
@kris-bosland  I copied the basic methods from the set() section, assuming that it was all there for a reason.

Reply:
With `set`, I think it makes sense to create a non-existent file, but with `delete`, I don't think it makes as much sense to try to create the config file if it doesn't already exist.